src/Symfony/Component/AssetMapper/ImportMap/ImportMapManager.php

```diff
@@ -0,0 +1,399 @@
+        $imports = $this->convertEntriesToImports($entries);
+
+        $importmap['imports'] = $imports;
+
+        // Use JSON_UNESCAPED_SLASHES | JSON_HEX_TAG to prevent XSS
+        $this->json = json_encode($importmap, \JSON_THROW_ON_ERROR | \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES | \JSON_HEX_TAG);
+    }
+
+    /**
+     * @param PackageRequireOptions[] $packagesToRequire
+     * @param string[] $packagesToRemove
+     * @return ImportMapEntry[]
+     */
+    private function updateImportMapConfig(bool $update, array $packagesToRequire, array $packagesToRemove): array
+    {
+        $currentEntries = $this->loadImportMapEntries();
+
+        foreach ($packagesToRemove as $packageName) {
+            if (!isset($currentEntries[$packageName])) {
+                throw new \InvalidArgumentException(sprintf('Package "%s" listed for removal was not found in "%s".', $packageName, basename($this->importMapConfigPath)));
+            }
+
+            $this->cleanupPackageFiles($currentEntries[$packageName]);
+            unset($currentEntries[$packageName]);
+        }
+
+        if ($update) {
+            foreach ($currentEntries as $importName => $entry) {
+                if (null === $entry->url) {
+                    continue;
+                }
+
+                // assume the import name === package name, unless we can parse
+                // the true package name from the URL
+                $packageName = $importName;
+                $registry = null;
+
+                // try to grab the package name & jspm "registry" from the URL
+                if (str_starts_with($entry->url, 'https://ga.jspm.io') && 1 === preg_match(self::PACKAGE_PATTERN, $entry->url, $matches)) {
+                    $packageName = $matches['package'];
+                    $registry = $matches['registry'] ?? null;
+                }
+
+                $packagesToRequire[] = new PackageRequireOptions(
+                    $packageName,
+                    null,
+                    $entry->isDownloaded,
+                    $entry->preload,
+                    $importName,
+                    $registry,
+                );
+
+                // remove it: then it will be re-added
+                $this->cleanupPackageFiles($entry);
+                unset($currentEntries[$importName]);
+            }
+        }
+
+        $newEntries = $this->requirePackages($packagesToRequire, $currentEntries);
+        $this->writeImportMapConfig($currentEntries);
+
+        return $newEntries;
+    }
+
+    /**
+     * Gets information about (and optionally downloads) the packages & updates the entries.
+     *
+     * Returns an array of the entries that were added.
+     *
+     * @param PackageRequireOptions[] $packagesToRequire
+     * @param ImportMapEntry[] $importMapEntries
+     */
+    private function requirePackages(array $packagesToRequire, array &$importMapEntries): array
+    {
+        if (!$packagesToRequire) {
+            return $importMapEntries;
```

Comment:
should this really return the current entries here ? They have not been added AFAICT.